### PR TITLE
test PR for M1 release support

### PR DESCRIPTION
**Pull request by nilslice**
This was _meant_ to be a PR to my fork for testing prior to opening one here 😆 . I guess I need more coffee...

In any case, this includes Rust and JS code changes to include support for installing `wrangler` via `npm`, as well as the `wrangler` dependencies when running `wrangler generate ...`. However, it does not include the automation to release pre-built binaries of those dependencies (e.g. `https://workers.cloudflare.com/get-binary/ashleygwilliams/cargo-generate/v0.5.0/x86_64-apple-darwin.tar.gz`) - I'm not sure from where those are built / released.

The actions workflow has been updated to include steps for the `aarch64-apple-darwin` target.

Also, this will likely fail for a few runs... just a starting point to figure out where things are happening.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 [package]
 name = "wrangler"
-version = "1.13.0"
-authors = ["The Wrangler Team <wrangler@cloudflare.com>", "Avery Harnish <averyharnish@gmail.com>", "Ashley Lewis <ashleymichal@gmail.com>", "Ashley Williams <ashley666ashley@gmail.com>", "Nat Davidson <davidson@cloudflare.com>"]
+version = "1.14.0"
+authors = ["The Wrangler Team <wrangler@cloudflare.com>", "Avery Harnish <averyharnish@gmail.com>", "Ashley Lewis <ashleymichal@gmail.com>", "Ashley Williams <ashley666ashley@gmail.com>", "Nat Davidson <davidson@cloudflare.com>", "Steve Manuel <nilslice@gmail.com>"]
```
Comment (xtuc) [inline Cargo.toml:4]: i'm not sure we add individual contributors to the author list?

Comment (koeninger) [reply Cargo.toml:4]: yeah if anything this should just be wrangler@cloudflare.com at this point

**Comment by Friss** [comment]
Hey just wanted to check in on this PR. It would be great to get it to land on master.

**Comment by nilslice** [comment]
hey @Friss - working on it this week, fingers crossed 😄

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -105,6 +106,13 @@ jobs:
           mv ${{ env.RELEASE_ADDS }} ./dist
           7z a -ttar -so -an ./dist | 7z a -si ./${{ env.RELEASE_DIR }}/${{ env.RELEASE_BIN }}-${{ steps.get_version.outputs.VERSION }}-${{ env.MACOS_TARGET }}.tar.gz
 
+      - name: Create tarball (MacOS M1)
```
Comment (Electroid) [inline .github/workflows/release.yml:109]: Instead of M1, should we just say `Apple Silicon`? Since I presume we'll soon see, M2, etc.

Comment (nilslice) [reply .github/workflows/release.yml:109]: good call!

Comment (xortive) [inline .github/workflows/release.yml:109]:
```suggestion
      - name: Create tarball (Apple Silicon)
```

**Review by xortive — APPROVED** [review]

**Pull request merged**